src/Provider/Provider.Web/Orchestrators/Part1/EmployerOrchestrator.cs

```diff
@@ -0,0 +1,96 @@
+
+namespace Esfa.Recruit.Provider.Web.Orchestrators.Part1
+{
+    public class EmployerOrchestrator : EntityValidatingOrchestrator<Vacancy, EmployersEditViewModel>
+    {
+        private const VacancyRuleSet ValidationRules = VacancyRuleSet.EmployerAccountId | VacancyRuleSet.TrainingProvider;
+        private readonly IProviderVacancyClient _providerVacancyClient;
+        private readonly IRecruitVacancyClient _recruitVacancyClient;
+
+        public EmployerOrchestrator(ILogger<EmployerOrchestrator> logger, 
+            IProviderVacancyClient providerVacancyClient, IRecruitVacancyClient recruitVacancyClient)
+            : base(logger)
+        {
+            _providerVacancyClient = providerVacancyClient;
+            _recruitVacancyClient = recruitVacancyClient;
+        }
+
+        protected override EntityToViewModelPropertyMappings<Vacancy, EmployersEditViewModel> DefineMappings()
+        {
+            var mappings = new EntityToViewModelPropertyMappings<Vacancy, EmployersEditViewModel>();
+
+            mappings.Add(e => e.EmployerAccountId, vm => vm.SelectedEmployerId);
+
+            return mappings;
+        }
+        public async Task<EmployersViewModel> GetEmployersViewModelAsync(VacancyRouteModel vacancyRouteModel)
+        {
+            var editVacancyInfo = await _providerVacancyClient.GetProviderEditVacancyInfoAsync(vacancyRouteModel.Ukprn);
+
+            var vm = new EmployersViewModel
+            {
+                PageInfo = new PartOnePageInfoViewModel(),
+                Employers = editVacancyInfo.Employers.Select(e => new EmployerViewModel {Id = e.Id, Name = e.Name})
+            };
+
+            return vm;
+        }
+
+        public async Task<OrchestratorResponse<Guid>> PostEmployerEditViewModelAsync(EmployersEditViewModel viewModel, VacancyUser user)
+        {
+            if (!viewModel.VacancyId.HasValue) // Create if it's a new vacancy
+            {
+                 var newVacancy = new Vacancy
+                {
+                    TrainingProvider = new TrainingProvider{Ukprn = viewModel.Ukprn},
+                    EmployerAccountId = viewModel.SelectedEmployerId,
+                    EmployerName = await GetEmployerNameAsync(viewModel.Ukprn, viewModel.SelectedEmployerId)
+                };
+
+                return await ValidateAndExecute(
+                    newVacancy, 
+                    v => _recruitVacancyClient.Validate(v, ValidationRules),
+                    async v => await _providerVacancyClient.CreateVacancyAsync(SourceOrigin.ProviderWeb, 
+                        viewModel.SelectedEmployerId, viewModel.Ukprn, user, UserType.Provider));
+            }
+
+            var vacancy = await Utility.GetAuthorisedVacancyForEditAsync(_providerVacancyClient, _recruitVacancyClient, 
+                new VacancyRouteModel{ Ukprn = viewModel.Ukprn, VacancyId = viewModel.VacancyId.GetValueOrDefault() }, 
```

Comment:
Recently I've been adding the VRM as an extra parameter to the controller action and passing it down to the orchestrator so I don't have to recreate it.

For example: https://github.com/SkillsFundingAgency/das-recruit/blob/c9d054253eb5566f9d31fcd59454ef10aa15a853/src/Employer/Employer.Web/Orchestrators/Part2/QualificationsOrchestrator.cs#L71

https://github.com/SkillsFundingAgency/das-recruit/blob/c9d054253eb5566f9d31fcd59454ef10aa15a853/src/Employer/Employer.Web/Controllers/Part2/QualificationsController.cs#L36